content/contributing/development/ui/scss.rst

```diff
@@ -0,0 +1,259 @@
+.. * `[modifier]` - optional modifier;
+
+.. .. code-block:: scss
+..     :caption: :file:`mail/static/src/scss/composer.scss`
+
+..     .o_mail_emoji {
+..     }
+
+.. .. code-block:: scss
+..     :caption: :file:`mail/static/src/components/chat_window/chat_window.scss`
+
+..     .o_ChatWindow {
+
+..         .o_ChatWindow_newMessageFormInput {
+..         }
+
+..         .o_ChatWindow_newMessageFormInput_active {
+..         }
+..     }
+
+
+.. Exceptions to these rules are `Bootstrap default classes <https://getbootstrap.com/docs/5.1/getting-started/introduction/>`_ and odoo utility-classes generated using `Bootstrap's API <https://getbootstrap.com/docs/5.1/utilities/api/>`_.
+
+
+
+Variables
+---------
+
+Global SCSS Variables
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Our standard convention is `$o-[root]-[element]-[property]-[modifier]`:
+
+* `[root]` - either the component **or** the module name (components take priority);
+* `[element]` - optional identifier for inner elements;
+* `[property]` - the property/behaviour defined by the variable;
+* `[modifier]` - optional modifier;
+
+.. code-block:: scss
+
+    $o-block-color: value;
+    $o-block-title-color: value;
+    $o-block-title-color-hover: value;
+
+.. note::
+   With Odoo, "Global" SCSS variables are actually accessible within the current bundle only.
+
+.. _scss/scoped-scss-var:
+
+Scoped SCSS Variables
+~~~~~~~~~~~~~~~~~~~~~
+
+These variables are declared within blocks and are not accessible from the ouside.
+Our standard convention is `$-[variable name]`.
+
+.. code-block::
+
+    .o_myComponent {
+        $-inner-gap: compute-something;
+
+        margin-right: $-inner-gap;
+
+        .o_myComponent_child {
+            margin-right: $-inner-gap * 0.5;
+        }
+    }
+
+.. seealso::
+   Variables scope on `SASS Documentation website <https://sass-lang.com/documentation/variables#scope>`_
+
+.. _scss/css-var:
+
+CSS Variables (aka "Custom properties")
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In odoo the use of CSS variables is striclty DOM related.
+Use them for **contextually** adapt design and layout.
+
+Our standard convention is BEM, so `--[root]__[element]-[property]--[modifier]`.
+
+* `[root]` - either the component **or** the module name (components take priority);
+* `[element]` - optional identifier for inner elements;
+* `[property]` - the property/behaviour defined by the variable;
+* `[modifier]` - optional modifier;
+
+
+.. code-block:: scss
+
+    .o_kanban_record {
+        --KanbanRecord-width: value;
+        --KanbanRecord__picture-border: value;
+        --KanbanRecord__picture-border--active: value;
+    }
+
+    // Adapt the component when rendered in another context.
+    .o_form_view {
+        --KanbanRecord-width: another-value;
+        --KanbanRecord__picture-border: another-value;
+        --KanbanRecord__picture-border--active: another-value;
+    }
+
+
+.. seealso::
+   Read more on :doc:`CSS variables <css_variables>` in odoo.
```

Comment:
```suggestion
   Read more on :doc:`CSS variables <css_variables>` in Odoo.
```
Not sure if that's really important, but I guess it's prettier with the uppercase, I'll let you decide and stop commenting those, you can find it easily